### PR TITLE
Add Resource Referral experiment

### PR DESCRIPTION
**Pull request by yoomlam**
Add Resource Referral experiment from private repo.

Description: https://github.com/navapbc/labs-gen-ai-experiments/wiki/Resource-Referral-Experiment

### CODE REVIEW

**Comment by yoomlam** [comment]
@ccheng26 Can you add instructions for any fulfilling any requirements needed to run the code, such as an `input(".env is missing! Populate it with ...")` or a `README.md` file?

File: 01-resource-referral/langgraph-workflow.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    def run_llms(self, state):
+        print("\nRUN_LLMS node", state["messages"][-1])
+        return None
+
+    def llm_211_api(self, state):
+        # find the last message that mentions the "user_query"
+        user_message = next(m for m in reversed(state["messages"]) if "user_query" in m)
+        print("\nLLM_211_API node", user_message)
+
+        # Human-in-the-loop example:
```
Comment (ccheng26) [inline 01-resource-referral/langgraph-workflow.py:137]: did we want to remove this?

Comment (yoomlam) [reply 01-resource-referral/langgraph-workflow.py:137]: I don't mind `Human-in-the-loop example` being there. What are your thoughts?

Comment (ccheng26) [reply 01-resource-referral/langgraph-workflow.py:137]: I'm open to having it there, but maybe we want to handle the "n" option differently as opposed to throwing a ValueError. I'm currently going through the comments and seeing where we can do some clean up

Comment (yoomlam) [reply 01-resource-referral/langgraph-workflow.py:137]: Gotcha, how about a call to `sys.exit()` instead of the `ValueError`?

**Review by ccheng26 — APPROVED** [review]
lgtm

**Pull request merged**